# Домашнє завдання до теми "Основи роботи з рядками та математичні операції"

1. Створіть змінні `a`, `b`, `c` та `d`, задайте їм значення відповідно 101, 210, 111, 402, та обчисліть з допомогою Python наступний вираз:  $$ \frac{(a+b) * c}{d} $$

**Важливо**: тут ми маємо обчислити значення саме з використанням змінних, а не просто чисел, як би ми робили у калькуряторі. Змінні потрібні аби за бажання ми могли змінити їх значення та не переписувати код обчислення виразу, бо він буде працювати для різних значень змінних.

In [1]:
a = 101
b = 210
c = 111
d = 402

In [3]:
(a + b) * c / d

85.8731343283582

Очікувана відповідь: 85.8731343283582

2. Якою буде остача від ділення числа 11283 на 89? Обчисліть з допомогою Python.

In [8]:
a = 11283
b = 89
g = a % b
print(g)

69


3. Скільки цілих разів поміститься число 89 в 11283? (Або що те саме: який буде результат цілочисельного ділення 11283 на 89?)

In [9]:
a = 11283
b = 89
g = a // b
print(g)

126


4. Запишіть у змінну `name` ваше імʼя.

In [11]:
name = "Pavlo"

5. Запишіть у змінну `surname` ваше прізвище.

In [12]:
surname = "Grechnyi"

6. Сконкатенуйте імʼя та прізвище в одну фразу через пробіл. Запишіть результат у нову змінну `name_surname` та виведіть результат на екран.  

In [14]:
name_surname = "Pavlo Grechnyi"
print(name_surname)

Pavlo Grechnyi


Очікувана відповідь (тільки з вашими даними): 'Hanna Pylieva'

7. Зробіть усі літери змінної `name_surname` заглавними використовуючи метод рядків в Python:

In [19]:
name_surname = s
s = "Pavlo Grechnyi"


Очікувана відповідь (тільки з вашими даними): 'HANNA PYLIEVA'

In [20]:
print(s.upper())


PAVLO GRECHNYI


8. Виведіть третю літеру вашого `name_surname`. Тут уважно, треба саме третя літера. Згадайте, з якого значення починається індексування значень в Python.

In [21]:
s[2]

'v'

Очікувана відповідь (тільки з вашими даними): 'n'

9. Виведіть першу та останню літеру вашого прізвища.

In [26]:
(s[6], s[-1])

('G', 'i')

Очікувана відповідь (тільки з вашими даними): ('P', 'a')

10. Виведіть довжину вашого прізвища.

In [29]:
surname = "Grechnyi"
len(surname)

8

Очікувана відповідь (тільки з вашими даними): 7


11. Виведіть кількість літер `a` у столиці Мадагаскару `Antananarivo`. Важливо врахувати всі літери, як великі, так і маленькі.

In [30]:
s = "Antananarivo"
s.lower().count("a")


4

Очікувана відповідь: 4

12. Створіть рядок `my_string` зі значенням: "The quick brown fox jumps over the lazy dog!!!".   
Перевірте, чи починається рядок `my_string` з "The" і чи закінчується він "dog!". Виведіть результати на екран.

In [38]:
my_string = "The quick brown fox jumps over the lazy dog!!!"

print(my_string.startswith("The"))
print(my_string.endswith("dog!"))


True
False


13. Розділіть рядок `my_string` на список окремих слів, а потім з'єднайте всі слова через нижнє підкреслювання. Виведіть результат на екран.

In [ ]:
print("_".join(["The", "quick", 'brown', "fox", "jumps", "over", "the", "lazy", "dog!!!"]))


'The_quick_brown_fox_jumps_over_the_lazy_dog!!!'

**Не зрозумів чому лапки все ж таки залишаються? пробував просто через "_".join, потім через print і вони все ж таки залишаються.**

Очікуваний результат: `The_quick_brown_fox_jumps_over_the_lazy_dog!!!`